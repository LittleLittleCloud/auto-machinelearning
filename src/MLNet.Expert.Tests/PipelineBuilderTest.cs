// <copyright file="PipelineBuilderTest.cs" company="BigMiao">
// Copyright (c) BigMiao. All rights reserved.
// </copyright>

using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using FluentAssertions;
using Microsoft.ML;
using Microsoft.ML.Data;
using MLNet.AutoPipeline;
using MLNet.Expert;
using Newtonsoft.Json;
using System.Linq;
using Xunit;
using Xunit.Abstractions;

namespace MLNet.Expert.Tests
{
    public class PipelineBuilderTest : TestBase
    {
        public PipelineBuilderTest(ITestOutputHelper helper)
            : base(helper)
        {
        }

        [Fact]
        [UseApprovalSubdirectory("ApprovalTests")]
        [UseReporter(typeof(DiffReporter))]
        public void PipelineBuilder_should_create_serializable_pipeline()
        {
            var context = new MLContext();
            var columns = new Column[]
            {
                new Column("cat1", ColumnType.Catagorical, ColumnPurpose.CategoricalFeature),
                new Column("numeric1", ColumnType.Numeric, ColumnPurpose.NumericFeature),
                new Column("text1", ColumnType.Numeric, ColumnPurpose.TextFeature),
                new Column("label", ColumnType.Catagorical, ColumnPurpose.Label),
            };

            var pb = new PipelineBuilder(TaskType.BinaryClassification, false, true);
            var pipeline = pb.BuildPipeline(context, columns);

            var pipelineDataContract = pipeline.ToDataContract();
            var json = JsonConvert.SerializeObject(pipelineDataContract, Formatting.Indented);
            Approvals.Verify(json);
            var restoredPipeline = pipelineDataContract.ToPipeline(context);
            var restoredJson = JsonConvert.SerializeObject(restoredPipeline.ToDataContract(), Formatting.Indented);
            restoredJson.Should().Be(json);
        }
    }
}
