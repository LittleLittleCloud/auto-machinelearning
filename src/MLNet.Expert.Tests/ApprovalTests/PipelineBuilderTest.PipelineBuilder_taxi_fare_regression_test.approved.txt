{
  "Estimators": [
    [
      {
        "EstimatorName": "OneHotEncodingEstimator",
        "InputColumns": [
          "vendor_id"
        ],
        "OutputColumns": [
          "vendor_id"
        ],
        "Scope": 7
      }
    ],
    [
      {
        "EstimatorName": "OneHotEncodingEstimator",
        "InputColumns": [
          "rate_code"
        ],
        "OutputColumns": [
          "rate_code"
        ],
        "Scope": 7
      }
    ],
    [
      {
        "EstimatorName": "MissingValueReplacingEstimator",
        "InputColumns": [
          "passenger_count"
        ],
        "OutputColumns": [
          "passenger_count"
        ],
        "Scope": 7
      }
    ],
    [
      {
        "EstimatorName": "MissingValueReplacingEstimator",
        "InputColumns": [
          "trip_time"
        ],
        "OutputColumns": [
          "trip_time"
        ],
        "Scope": 7
      }
    ],
    [
      {
        "EstimatorName": "MissingValueReplacingEstimator",
        "InputColumns": [
          "trip_distance"
        ],
        "OutputColumns": [
          "trip_distance"
        ],
        "Scope": 7
      }
    ],
    [
      {
        "EstimatorName": "MissingValueReplacingEstimator",
        "InputColumns": [
          "payment_type"
        ],
        "OutputColumns": [
          "payment_type"
        ],
        "Scope": 7
      }
    ],
    [
      {
        "EstimatorName": "ColumnConcatenatingEstimator",
        "InputColumns": [
          "vendor_id",
          "rate_code",
          "passenger_count",
          "trip_time",
          "trip_distance",
          "payment_type"
        ],
        "OutputColumns": [
          "_FEATURE"
        ],
        "Scope": 7
      }
    ],
    [
      {
        "EstimatorName": "GamRegressionTrainer",
        "InputColumns": [
          "fare_amount",
          "_FEATURE"
        ],
        "OutputColumns": [
          "Score"
        ],
        "Scope": 7
      },
      {
        "EstimatorName": "LbfgsPoissonRegressionTrainer",
        "InputColumns": [
          "fare_amount",
          "_FEATURE"
        ],
        "OutputColumns": [
          "Score"
        ],
        "Scope": 7
      }
    ]
  ]
}